js/src/pages/settings/enhanced-conversions/useEnableEnhancedConversions.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useAppDispatch } from '~/data';
+import useAppSelectDispatch from '~/hooks/useAppSelectDispatch';
+
+const selectorName = 'getEnableEnhancedConversions';
+
+/**
+ * Custom hook to manage the state of enhanced conversions.
+ * @return {Array} An array containing the current state and a function to toggle it.
+ */
+export const useEnableEnhancedConversions = () => {
+	const { data, isResolving } = useAppSelectDispatch( selectorName );
```

Comment:
@ankitrox Why we are using `useAppSelectDispatch` here? 🤔

Reply:
I'am thinking that the selector should just use `useSelect` and return the value and then within `SetupEnhancedConversions`, dispatch `updateEnhancedConversionsStatus` within the `handleOnChange` function.